Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/BlendGraphWidget.cpp

```diff
@@ -1243,6 +1243,15 @@ namespace EMStudio
         {
             AZ_Error("EMotionFX", false, outResult.c_str());
         }
+
+        for (int i = 0; i < animGraph->GetNumNodeGroups(); i++)
+        {
+            if (animGraph->GetNodeGroup(i)->GetNumNodes() == 0)
+            {
+                animGraph->RemoveNodeGroup(i);
+                i--;
+            }
+        }
```

Comment:
suggestion: A slight alternative to this could be...

```
for (int i = 0; i < animGraph->GetNumNodeGroups();)
{
    if (animGraph->GetNodeGroup(i)->GetNumNodes() == 0)
    {
        animGraph->RemoveNodeGroup(i);
    }
    else
    {
        i++;
    }
}
```

Which is maybe a little clear but not a huge deal

Reply:
(not sure how node groups are stored under the hood but this can also be an n^2 operation if there are a lot of node groups to remove and they're stored in a vector, so an erase/remove idiom can often be waaaay faster in practice - API might make it hard to do here but just wanted to mention it as something to consider)

Reply:
It is like you said. Could be worth implementing it in the animgraph itself entirely (something like `AnimGraph::ClearEmptyNodeGroups`). Just didn't want to introduce too much extra API , but I can do that here too if you like.

Reply:
It's unlikely that the number of groups to be deleted is large. In the unlikely worst case, it would be equal to the number of nodes being assigned to a new group. More commonly it is probably only 1 group that needs to be deleted.

Reply:
Yeah this is fine as-is, sorry more just musing out loud, thanks for checking